File: C - C++/TowerOfHanoi/TowerOfHanoi.cpp
// C++ recursive function to  
// solve tower of hanoi puzzle  
#include <bits/stdc++.h> 
using namespace std; 
  
static long int counter = 0;
  
void towerOfHanoi(int n, char from_rod, 
                    char to_rod, char aux_rod)  
{  
    counter++;
    if (n == 1)  
    {  
        cout << "Move disk 1 from " << from_rod <<  
                            " to " << to_rod<<endl;  
        return;  
    }  
    towerOfHanoi(n - 1, from_rod, aux_rod, to_rod);  
    cout << "Move disk " << n << " from " << from_rod << 
                                " to " << to_rod << endl;  
    towerOfHanoi(n - 1, aux_rod, to_rod, from_rod);  
}  
  
// Driver code 
int main()  
{  
    int n; // Number of disks
    cin>>n;
    char A,B,C;
    cin>>A>>B>>C;
    towerOfHanoi(n, A, C, B); // name of labels are assigned to A, B and C 
    cout<<"It took "<<counter<<" moves in total"<<endl;
    return 0;  
}

File: Python/Tower_of_Hanoi/tower_of_hanoi.py
"""This program displays the moves to solve
    Tower of Hanoi puzzle"""


def toh(n, source, aux, target):
    if n == 1:
        print(f'Move Disk 1 from {source} to {target}')
        return 1
    left_count = toh(n - 1, source, target, aux)
    print(f'Move Disk {n} from {source} to {target}')
    right_count = toh(n - 1, aux, source, target)
    return left_count + right_count + 1


if __name__ == "__main__":
    n, source, aux, target = input(
        "Number of disks and 3 towers labels: ").split()
    n = int(n)
    count = toh(n, source, aux, target)
    print(f"It took {count} moves in total")
